plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -354,7 +370,44 @@ maybeAddSubst a s n rhs = do
             let acceptable = costIsAcceptable t && sizeIsAcceptable t
             -- See Note [Inlining and purity]
             termIsPure <- checkPurity t
-            pure $ acceptable && case s of { Strict -> termIsPure; NonStrict -> True; }
+
+            pure $ acceptable && termIsPure
```

Comment:
Why not also check `immediatelyEvaluated` here? Could it be that `immediatelyEvaluated` is true, but usage count > 1, and so `preInlineUnconditional` refuses to inline it?

Reply:
Right, and that case is unsafe! Because then we could duplicate the effects (or remove them, if it's dead). The note mentions this, but I'll call it out explicitly here.

Reply:
I thought effect = looping or error, both of which terminates the evaluation. So as long as the it is `immediatelyEvaluated`, it doesn't matter whether it is evaluated more times?

But then if a binding is impure, its size is probably unacceptable in the first place, so I guess it doesn't matter.

Reply:
Ah, right.